src/main/java/com/mineboundteam/minebound/entity/RockProjectile.java

```diff
@@ -0,0 +1,129 @@
+package com.mineboundteam.minebound.entity;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.particles.ParticleTypes;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.projectile.AbstractArrow;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.EntityHitResult;
+import net.minecraft.world.phys.Vec3;
+
+
+import java.util.List;
+
+import static com.mineboundteam.minebound.entity.registry.EntityRegistry.ROCK_PROJECTILE;
+
+
+/**
+ * Derived from {@link FireProjectile}
+ * Derived from {@link net.minecraft.world.entity.projectile.AbstractArrow}
+ */
+public class RockProjectile extends AbstractArrow {
+    private double damage;
+    private double originalX;
+    private double originalY;
+    private double originalZ;
+    private int maxDist;
+    private float damageRadius;
+    public RockProjectile(EntityType<? extends RockProjectile> pEntityType, Level pLevel) {
+        super(pEntityType, pLevel);
+        //Adds trailing particles
+        this.setCritArrow(true);
+        this.setSoundEvent(SoundEvents.STONE_BREAK);
+        //Overwrite damage that would have been applied in AbstractArrow
+        this.setBaseDamage(0.0D);
+    }
+
+    public RockProjectile(Level pLevel, LivingEntity shooter, double pX, double pY, double pZ, double damage, float damageRadius, int maxDistance) {
+        super(ROCK_PROJECTILE.get(), pX, pY, pZ, pLevel);
+        this.setOwner(shooter);
+
+        this.damage = damage;
```

Comment:
`damage` has to be cast to `float` every time it is used, should just change it's type to `float`.